File: java/src/tk/csparpa/gofdp/abstractfactory/ConcreteFactory.java
package tk.csparpa.gofdp.abstractfactory;

/**
 * Concrete implementation for AbstractFactory class: this class implements the
 * creational methods for all the products of a specific family
 */
public class ConcreteFactory implements AbstractFactory {

	@Override
	public AbstractProductX createProductX() {
		System.out.println("Called: implementation of abstract factory; "
				+ "will return: ConcreteProductX instance");
		return new ConcreteProductX();
	}

	@Override
	public AbstractProductY createProductY() {
		System.out.println("Called: implementation of abstract factory; "
				+ "will return: ConcreteProductY instance");
		return new ConcreteProductY();
	}
}

File: python/abstractfactory/concrete_factory.py
from abstract_factory import AbstractFactory
from concrete_product_X import ConcreteProductX
from concrete_product_Y import ConcreteProductY

class ConcreteFactory(AbstractFactory):
    """
    Concrete implementation for AbstractFactory class: this class implements the
    creational methods for all the products of a specific family
    """
    def create_product_X(self):
        print("Called: implementation of abstract factory; will return: " + \
              "ConcreteProductX instance")
        return ConcreteProductX()

    def create_product_Y(self):
        print("Called: implementation of abstract factory; will return: " + \
              "ConcreteProductY instance")
        return ConcreteProductY()